### PR TITLE
Add infiniband support.

### PR DESCRIPTION
**Pull request by orviz**
+ Through flavor's extra_specs parsing.
+ Metadata key&value is specified under:
    templates>extra_specs>infiniband
  in configuration file.
+ Only if key&value match with the values in the configuration,
  the infiniband support is published.
+ Currently available through GLUE2ExecutionEnvironmentNetworkInfo.

### CODE REVIEW

File: etc/sample.openstack.yaml [hunk 1]
```diff
@@ -37,9 +37,12 @@ compute:
         defaults:
             platform: amd64
             network: public
+            extra_specs:
```
Comment (alvarolopez) [inline etc/sample.openstack.yaml]: This should be removed, as the static information and its defaults are going to be added if there is no such extra_spec in the flavor (i.e. by default all flavors would have infiniband!)

File: cloud_info/providers/openstack.py [hunk 2]
```diff
@@ -281,10 +286,14 @@ def get_templates(self):
                 flavor_id = str(getattr(flavor, flavor_id_attr))
                 template_id = '%s%s#%s' % (URI, tpl_sch,
                                            OpenStackProvider.occify(flavor_id))
+                flavor_ib_v = flavor.get_keys().get(flavor_ib_k, False)
+                template_ib = (True if flavor_ib_v == extra_specs_ib['value']
```
Comment (alvarolopez) [inline cloud_info/providers/openstack.py]: This line can be changed to `template-ib = flavor_ib_v == extra_specs_ib['value']` (the rest is not needed)

**Review by alvarolopez — CHANGES_REQUESTED** [review]

**Comment by orviz** [comment]
@alvarolopez @aidaph the requested changes have been addressed. The new command line opts, `extra-specs-infiniband-key` and  `extra-specs-infiniband-value`, allow easier provision of required info

**Comment by orviz** [comment]
Just realized this PR is to be merged in master while it should be done against DEEP branch

**Pull request closed without merging**